### ISSUE TITLE
React.cloneElement facade missing

### ISSUE THREAD

**Issue by cquiroz** [japgolly/scalajs-react#477]
I'm trying to port a component that uses `cloneElement` but there is no facade for it.

Would it make sense to add it? 

The component uses it to add an event handler to the children
`React.cloneElement(elem, {...props, onClick: this.onClick});`

**Comment by cquiroz** [comment]
I'm coming up with these signatures, do they seem right?

```
  final def cloneElement(element: Element                                   ): Element = js.native
  final def cloneElement(element: Element, props: js.Object                 ): Element = js.native
  final def cloneElement(element: Element, props: js.Object, children: Node*): Element = js.native
```

**Comment by cquiroz** [comment]
I'd like `cloneElement` to return `DomElement` to access `PropWithChildren` but I'm not sure if that would be a correct return for `cloneElement`

**Comment by japgolly** [comment]
Closed via b887aae2dd0a4b1fe6b2607acaebf0199415fe28
